Framework/Parallel/src/IO/Chunker.cpp

```diff
@@ -0,0 +1,220 @@
+#include <numeric>
+
+#include "MantidParallel/Communicator.h"
+#include "MantidParallel/IO/Chunker.h"
+
+namespace Mantid {
+namespace Parallel {
+namespace IO {
+
+namespace {
+/// Helper to build partition (subgroup of ranks with subgroup of banks).
+std::pair<int, std::vector<size_t>>
+buildPartition(const int totalWorkers, const size_t totalSize,
+               std::vector<std::tuple<size_t, size_t, bool>> &sortedSizes) {
```

Comment:
I appreciate that this is a internal linkage free function, but a little hard to understand what for example sortedSizes type without looking at rest of code. I'm not suggesting changing the data structures at all.

Reply:
solution suggested below